libmamba/src/core/subdirdata.cpp

```diff
@@ -10,103 +10,154 @@
 #include "mamba/core/output.hpp"
 #include "mamba/core/package_cache.hpp"
 #include "mamba/core/subdirdata.hpp"
+#include "mamba/core/thread_utils.hpp"
 #include "mamba/util/string.hpp"
 #include "mamba/util/url_manip.hpp"
 
-#include "progress_bar_impl.hpp"
-
 namespace mamba
 {
-    void subdir_metadata::serialize_to_stream(std::ostream& out) const
+    /*******************
+     * MSubdirMetadata *
+     *******************/
+
+    namespace
     {
-        nlohmann::json j;
-        j["url"] = url;
-        j["etag"] = etag;
-        j["mod"] = mod;
-        j["cache_control"] = cache_control;
-        j["size"] = stored_file_size;
+#ifdef _WIN32
+        std::chrono::system_clock::time_point filetime_to_unix(const fs::file_time_type& filetime)
+        {
+            // windows filetime is in 100ns intervals since 1601-01-01
+            static constexpr auto epoch_offset = std::chrono::seconds(11644473600ULL);
+            return std::chrono::system_clock::time_point(
+                std::chrono::duration_cast<std::chrono::system_clock::duration>(
+                    filetime.time_since_epoch() - epoch_offset
+                )
+            );
+        }
+#endif
 
-        auto nsecs = std::chrono::duration_cast<std::chrono::nanoseconds>(
-            stored_mtime.time_since_epoch()
-        );
-        j["mtime_ns"] = nsecs.count();
+        // parse json at the beginning of the stream such as
+        // "_url": "https://conda.anaconda.org/conda-forge/linux-64",
+        // "_etag": "W/\"6092e6a2b6cec6ea5aade4e177c3edda-8\"",
+        // "_mod": "Sat, 04 Apr 2020 03:29:49 GMT",
+        // "_cache_control": "public, max-age=1200"
+        std::string extract_subjson(std::ifstream& s)
```

Comment:
Do we know for sure these will be at the top of the json?
Once again something at the repodata.json needs to be updated so that we don't need to do this sort of things.

Reply:
> Do we know for sure these will be at the top of the json?

Nope, but I think it has been the case for a while. Besides, I took this code from the previous subdirdata, because redesigning the repodata.json file is beyond the scope of this PR. But I agree that we should change this to have something more robust.